compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -783,15 +783,7 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                 }
             }
             Abi::ScalarPair(a_layout, b_layout) => {
-                // There is no `rustc_layout_scalar_valid_range_start` for pairs, so
-                // we would validate these things as we descend into the fields,
-                // but that can miss bugs in layout computation. Layout computation
-                // is subtle due to enums having ScalarPair layout, where one field
-                // is the discriminant.
-                if cfg!(debug_assertions)
```

Comment:
Why is now the right time to always check this?

Reply:
Because the comment was wrong. `rustc_scalar_layout_valid` already applies to scalar pairs and that can actually be observed with `NonNull<dyn Trait>` types.

Reply:
AH, good catch. Would be good to have a test. :)